app/services/name_matcher.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class NameMatcher
```

Comment:
Nice! The class name might be a little ambiguous but I can't think of anything better